src/wallet.rs

```diff
@@ -670,3 +716,66 @@ impl FromStr for Chain {
 #[derive(Debug, thiserror::Error)]
 #[error("Unsupported chain: {0}")]
 pub struct WrongChain(String);
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct EncryptedWallet {
+    name: String,
+    encrypted_xprv: Vec<u8>,
+    sk_salt: [u8; 32],
+    chain: Chain,
+}
```

Comment:
Nice idea using serde for this! Can we have a snapshot test of the serialization to make sure we don't accidentially break the format?